### PR TITLE
polishtracker: add imdb search and code clean up

### PR DESCRIPTION
**Pull request by ngosang**
Tested. Part of #4859

### CODE REVIEW

**Comment by ngosang** [comment]
@RoloSoze ping

File: src/Jackett.Common/Indexers/PolishTracker.cs [hunk 1]
```diff
@@ -16,21 +18,17 @@ namespace Jackett.Common.Indexers
...
             "https://polishtracker.net/",
-            };
+        };
 
-        private new ConfigurationDataBasicLoginWithEmail configData
-        {
-            get => (ConfigurationDataBasicLoginWithEmail)base.configData;
-            set => base.configData = value;
-        }
+        private new ConfigurationDataBasicLoginWithEmail configData => (ConfigurationDataBasicLoginWithEmail)base.configData;
```
Comment (cadatoiva) [inline src/Jackett.Common/Indexers/PolishTracker.cs]: Does this work when changing the config values (delete and re-add)? I've been leaving set in for that purpose.

Comment (ngosang) [reply src/Jackett.Common/Indexers/PolishTracker.cs]: Yest, it works.

Comment (cadatoiva) [reply src/Jackett.Common/Indexers/PolishTracker.cs]: huh. Cool.

File: src/Jackett.Common/Indexers/PolishTracker.cs [hunk 2]
```diff
@@ -83,105 +83,87 @@ protected override async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuer
         {
             var releases = new List<ReleaseInfo>();
 
-            var searchUrl = TorrentApiUrl;
-            var searchString = query.GetQueryString();
-            var queryCollection = new List<KeyValuePair<string, string>>
+            var qc = new List<KeyValuePair<string, string>> // NameValueCollection don't support cat[]=19&cat[]=6
```
Comment (cadatoiva) [inline src/Jackett.Common/Indexers/PolishTracker.cs]: See #7915 for a fix for this. I was already working on it for my own refactoring.

**Comment by ngosang** [comment]
@cadatoiva I'm going to merge, we have to open another PR after #7915 to replace `List<KeyValuePair<string, string>>` in other trackers too.

**Pull request merged**